nautobot/ipam/models.py

```diff
@@ -1079,30 +1097,26 @@ def clean(self):
         if self.type == choices.IPAddressTypeChoices.TYPE_SLAAC and self.ip_version != 6:
             raise ValidationError({"type": "Only IPv6 addresses can be assigned SLAAC type"})
 
+        # Validate `parent` can be used as the parent for this ipaddress
+        if self.address and self.parent:
+            closes_parent = self._get_closest_parent()
+            if self.parent != closes_parent:
+                raise ValidationError({"parent": f"{self.parent} cannot be assigned as a parent."})
+            self.parent = closes_parent
+
         # Force dns_name to lowercase
         self.dns_name = self.dns_name.lower()
 
     def save(self, *args, **kwargs):
-        if not self.present_in_database:
-            if self._namespace is None:
-                raise ValidationError({"parent": "Namespace could not be determined."})
-            namespace = self._namespace
-        else:
-            namespace = self._namespace or self.parent.namespace
-
-        # Determine the closest parent automatically based on the Namespace.
-        self.parent = (
-            Prefix.objects.filter(namespace=namespace)
-            # 3.0 TODO: disallow IPAddress from parenting to a TYPE_POOL prefix, instead pick closest TYPE_NETWORK
-            # .exclude(type=choices.PrefixTypeChoices.TYPE_POOL)
-            .get_closest_parent(self.host, include_self=True)
-        )
-
         # 3.0 TODO: uncomment the below to enforce this constraint
         # if self.parent.type != choices.PrefixTypeChoices.TYPE_NETWORK:
         #     err_msg = f"IP addresses cannot be created in {self.parent.type} prefixes. You must create a network prefix first."
         #     raise ValidationError({"address": err_msg})
 
+        # validated_save is not always called especially in Test environment;
+        # but we still need to set the `self.parent` attr in these cases
+        if self.address and not self.parent:
+            self.parent = self._get_closest_parent()
```

Comment:
Does this mean that I can now do something like `IPAddress.objects.create(address="10.1.1.1", parent=Prefix.objects.get(network="172.16.0.0/16"))` and as long as I never call clean() it'll be accepted by the DB? We definitely should still prevent that.

Reply:
This would means we would still have to silently discard invalid parent, which is what the issue is reporting against.
Except we force validation save for `IPAddress`